lib/chef/knife/mixin/helper.rb

```diff
@@ -66,10 +66,10 @@ def validate_json(json)
 
       # I/P: String
       # O/P: true/false
-      # returns true if string is free of non-printable characters (escape sequences)
+      # returns true if string is free of non-printable characters (escape sequences) or includes '\n' or '\r', which is perfectly valid
       # this returns false for whitespace escape sequences as well, e.g. \n\t
       def printable?(string)
-        /[^[:print:]]|[[:space:]]/.match(string)
+        /[[:print:]]|[\n\r]/.match?(string)
```

Comment:
This isn't right either, it only ensures there's at least one printable character.

I think the regex just needs to check for any non-printable characters like this:

```ruby
/[^[:print:]]/.match?(string)
```